projects/alembic/alembic_dump_info_fuzzer.cc

```diff
@@ -0,0 +1,263 @@
+  const ISubDSchema &schema = mesh.getSchema();
+
+  dumpSubDAttributes(schema);
+  schema.getSubdivisionSchemeProperty();
+  schema.getFaceVaryingInterpolateBoundaryProperty();
+  schema.getFaceVaryingPropagateCornersProperty();
+  schema.getInterpolateBoundaryProperty();
+}
+
+void dumpFaceSet(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IFaceSet faceSet(node.getParent(), header.getName());
+  const IFaceSetSchema &schema = faceSet.getSchema();
+  schema.getNumSamples();
+}
+
+void dumpCurves(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const ICurves curves(node.getParent(), header.getName());
+  const ICurvesSchema &schema = curves.getSchema();
+
+  dumpCurvesAttributes(schema);
+}
+
+void dumpXform(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IXform xform(node.getParent(), header.getName());
+  const IXformSchema &schema = xform.getSchema();
+
+  schema.getNumSamples();
+  schema.getNumOps();
+}
+
+void dumpMaterial(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  IMaterial material(node.getParent(), header.getName());
+  IMaterialSchema &schema = material.getSchema();
+
+  std::vector<std::string> targetNames;
+  size_t targetCount = targetNames.size();
+
+  for (size_t t = 0; t < targetCount; t++) {
+    std::vector<std::string> shaderTypes;
+    schema.getShaderTypesForTarget(targetNames[t], shaderTypes);
+    const size_t shaderTypeCount = shaderTypes.size();
+    for (size_t s = 0; s < shaderTypeCount; s++) {
+
+      ICompoundProperty parameters =
+          schema.getShaderParameters(targetNames[t], shaderTypes[s]);
+      const size_t parameterCount = parameters.getNumProperties();
+    }
+  }
+}
+
+void dumpNodes(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  // Dump the general node information.
+  header.getName();
+  header.getFullName();
+  header.getMetaData().serialize();
+
+  // Dump the type specific information.
+  if (Alembic::AbcGeom::IPolyMesh::matches(header)) {
+    dumpPolyMesh(node);
+  } else if (Alembic::AbcGeom::ISubD::matches(header)) {
+    dumpSubD(node);
+  } else if (Alembic::AbcGeom::IFaceSet::matches(header)) {
+    dumpFaceSet(node);
+  } else if (Alembic::AbcGeom::ICurves::matches(header)) {
+    dumpCurves(node);
+  } else if (Alembic::AbcGeom::IXform::matches(header)) {
+    dumpXform(node);
+  } else if (Alembic::AbcMaterial::IMaterial::matches(header)) {
+    dumpMaterial(node);
+  } else { // Miscellaneous nodes such as the root.
+    ;
+  }
+
+  // Dump the child headers.
+  const size_t childCount = node.getNumChildren();
+  for (size_t i = 0; i < childCount; i++) {
+    dumpNodes(node.getChild(i));
+  }
+}
+
+void dumpInfo(const char *file) {
+  // Load the Alembic archive and verify that it is valid.
+  IFactory factory;
+  IArchive archive = factory.getArchive(file);
+
+  bool fileValid = archive.valid();
+
+  if (fileValid) {
```

Comment:
Stylistic nit: I don't see a reason to extract `fileValid` into its own variable; `if (archive.valid())` is perfectly clear.